plugins/coverimage.koplugin/main.lua

```diff
@@ -0,0 +1,177 @@
+local Device = require("device")
+
+if not Device.isAndroid() and not Device.isEmulator() then
+    return { disabled = true }
+end
+
+local DocSettings = require("docsettings")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local logger = require("logger")
+local _ = require("gettext")
+local T = require("ffi/util").template
+
+local CoverImage = WidgetContainer:new{
+    name = 'coverimage',
+    is_doc_only = true,
+    settings_id = 0,
+}
+
+function CoverImage:_enabled()
+    return self.enabled
+end
+
+function CoverImage:_restore()
+    return self.enabled and self.restore
+end
+
+function CoverImage:init()
+    self.cover_image_path = G_reader_settings:readSetting("cover_image_path") or "Cover.png"
+    self.enabled = G_reader_settings:isTrue("cover_image_enabled")
+    self.restore = G_reader_settings:isTrue("cover_image_restore") and self.enabled
+    self.ui.menu:registerToMainMenu(self)
+end
+
+function CoverImage:cleanUpImage()
+    local lfs = require("libs/libkoreader-lfs")
+    local has_bak = lfs.attributes(self.cover_image_path .. ".bak")
+    -- Delete image if backup exists, the backup could contain a user defined sleep image
+    if lfs.attributes(self.cover_image_path) and has_bak then
+        os.remove(self.cover_image_path)
+    end
+    -- Restore backup if it exists, so we can use the last image.
+    -- On Tolino a user defined /sdcard/suspend_others.jgp can be used as screensaver on system sleep.
+    if has_bak then
+        os.rename(self.cover_image_path .. ".bak", self.cover_image_path)
+    end
```

Comment:
Looks like all this can be wrap in a single `if has_bak then`.
(And may be have `self.cover_image_path .. ".bak"` in a local variable to avoid doing the concat twice.

About the last comment:
> On Tolino a user defined /sdcard/suspend_others.jgp can be used as screensaver on system sleep

You mean one could chose to not have a main image (the one that becomes .bak)? In this case, shouldn't you also `os.remove(self.cover_image_path)` when `has_bak` is false ?

Reply:
Thank you, the has_bak on the remove is wrong.